### PR TITLE
Refactor fuzzy search query

### PR DESCRIPTION
**Pull request by anthonyhashemi**
## Changes in this PR

Refactor fuzzy search query:

- paginate in the search route, not the query building function
- added tests for fuzzy query edge cases

## JIRA ticket
https://national-archives.atlassian.net/browse/AYR-574

### CODE REVIEW

File: app/tests/test_queries.py [hunk 1]
```diff
@@ -1,210 +1,281 @@
...
-                "test_file9.txt",
-                files[8].consignment.series.body.BodyId,
-                files[8].consignment.series.SeriesId,
-            ),
-        ]
-        assert result.pages > 0
-        assert result.has_next is False
-        assert result.has_prev is True
+
+#     # def test_build_fuzzy_search_query_multiple_matches(self):
```
Comment (colinbowen) [inline app/tests/test_queries.py]: Possible to remove these if they're not needed?

**Review by colinbowen — APPROVED** [review]
PR looks good just the commented out tests which could be removed

File: app/tests/test_queries.py (hunk 1 above)
Comment (anthonyhashemi) [reply app/tests/test_queries.py]: ah yeah thanks for spotting these, I had put them as placeholders. will remove for now to get unblocked, can add more later

**Review by pgandhizaizi — APPROVED** [review]
looks good
